sklearn/ensemble/iforest.py

```diff
@@ -116,7 +127,7 @@ class IsolationForest(BaseBagging):
     def __init__(self,
                  n_estimators=100,
                  max_samples="auto",
-                 contamination=0.1,
+                 contamination="auto",
```

Comment:
This looks like a backwards-incompatible change. Is that correct? If so, ideally we avoid the backwards incompatible change, perhaps by having a 'legacy' default setting meaning 0.1 with warning, which will change to 'auto' after a deprecation period. If you feel 0.1 was never a sensible choice, we at least need documentation under the changed models heading of the change log.

Reply:
Actually it looks like you're still maintaining the 0.1, at least when setting `_threshold_`. What's that doing?

Reply:
You're right this is a backwards-incompatible change, as `_threshold_` is here just for the deprecation warning and is not used anywhere. `0.1` was an arbitrary choice for the outlier proportion and was not very sensible given that the original paper gives a better heuristic to fix the decision function threshold. Do you think we need a deprecation period?